### PR TITLE
updated tests to ensure hex values are returned

### LINKED ISSUE #3563: test hex results of engine_exchangeTransitionConfigurationV1
### Description
Our implementation accepts already hexadecimal and integer as request. But it would be good to change EngineExchangeTransitionConfigurationTest to only use hexadecimals, to make sure that it continues to work in the future. Additionally we need to check that the response produces hex decimals as well.

### Acceptance Criteria
* Tests cover both hex and decimal values.

### PR DESCRIPTION
**Pull request by jflo**
Signed-off-by: Justin Florentine <justin+github@florentine.us>

## PR description
Uses Quantity for consistent formatting.

## Fixed Issue(s)
fixes #3563 

## Changelog

- [X] I thought about the changelog and included a [changelog update if required](https://wiki.hyperledger.org/display/BESU/Changelog).

### CODE REVIEW

**Review by macfarla — APPROVED** [review]

File: ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineExchangeTransitionConfigurationTest.java [hunk 1]
```diff
@@ -153,17 +153,33 @@ public void shouldAlwaysReturnResultsInHex() throws JsonProcessingException {
...
-    assertThat(mockResult.getTerminalTotalDifficultyAsString())
-        .isEqualTo(Difficulty.ZERO.toHexString());
+    assertThat(mockResult.getTerminalTotalDifficultyAsString()).isEqualTo("0x0");
     assertThat(mockResult.getTerminalBlockHashAsString()).isEqualTo(Hash.ZERO.toHexString());
 
     String json = mapper.writeValueAsString(mockResult);
     var res = mapper.readValue(json, Map.class);
     assertThat(res.get("terminalBlockNumber")).isEqualTo("0x0");
     assertThat(res.get("terminalBlockHash"))
         .isEqualTo("0x0000000000000000000000000000000000000000000000000000000000000000");
```
Comment (daniellehrner) [inline ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineExchangeTransitionConfigurationTest.java:163]: The leading 0s need to be stripped. The spec says:
```
Values of a field of QUANTITY type MUST be encoded as a hexadecimal string with a 0x prefix and the leading 0s stripped (except for the case of encoding the value 0) matching the regular expression ^0x(?:0|(?:[a-fA-F1-9][a-fA-F0-9]*))$.
```

File: ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineExchangeTransitionConfigurationTest.java [hunk 2]
```diff
@@ -153,17 +153,33 @@ public void shouldAlwaysReturnResultsInHex() throws JsonProcessingException {
...
+
+    assertThat(mockResult.getTerminalBlockNumberAsString()).isEqualTo("0x64");
+    assertThat(mockResult.getTerminalTotalDifficultyAsString()).isEqualTo("0x0");
+    assertThat(mockResult.getTerminalBlockHashAsString()).isEqualTo(Hash.ZERO.toHexString());
+
+    String json = mapper.writeValueAsString(mockResult);
+    var res = mapper.readValue(json, Map.class);
+    assertThat(res.get("terminalBlockNumber")).isEqualTo("0x64");
+    assertThat(res.get("terminalBlockHash"))
         .isEqualTo("0x0000000000000000000000000000000000000000000000000000000000000000");
```
Comment (daniellehrner) [inline ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineExchangeTransitionConfigurationTest.java:181]: Same as in the other comment. We need to strip the leading 0s for terminalBlockHash as well.

**Review by daniellehrner — CHANGES_REQUESTED** [review]

File: ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineExchangeTransitionConfigurationTest.java (hunk 1 above)
Comment (jflo) [reply ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineExchangeTransitionConfigurationTest.java:163]: terminalBlockHash is of type `DATA`, not `QUANTITY`. leading zeros are ok in those.

https://github.com/ethereum/execution-apis/blob/main/src/engine/specification.md#TransitionConfigurationV1

Comment (daniellehrner) [reply ethereum/api/src/test/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineExchangeTransitionConfigurationTest.java:163]: My bad. Then it's correct of course

**Review by daniellehrner — APPROVED** [review]

**Pull request merged**